notebooks/ullyses-walkthrough.ipynb

```diff
@@ -311,58 +392,228 @@
+    "print(cspec_filename)\n",
+    "with fits.open(cspec_filename) as cspec_hdu:\n",
+    "    print(cspec_hdu.info())\n",
+    "    print()\n",
+    "    print(cspec_hdu[1].columns)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# For each target, plot the level 4 preview-spec and the level 2 cspec products\n",
+    "for targ in targets:\n",
+    "    # use the built in ULLYSES function to find the HLSP name for the target\n",
+    "    ull_targ = match_aliases(targ).lower()\n",
+    "    \n",
+    "    # Create a variable to point to the correct preview-spec file\n",
+    "    prevspec_file = glob.glob(f'*{ull_targ}*_preview-spec.fits')[0]\n",
+    "    spec_file = Path(prevspec_file)\n",
+    "\n",
+    "    # preview-spec data\n",
+    "    wavelength, flux, error, snr, gratings = get_non_variable_data(spec_file)  \n",
+    "\n",
+    "    # Make an array to hold the level 2 filenames\n",
+    "    level2 = glob.glob(f'*{ull_targ}*cspec*')\n",
+    "    \n",
+    "    # Set up the plot - start with plotting the level 4 products\n",
+    "    fig = plt.figure()\n",
+    "    gs = fig.add_gridspec(3, hspace=0)\n",
+    "    axs = gs.subplots(sharex=True, sharey=False)\n",
+    "\n",
+    "    axs[0].plot(wavelength, flux, color='k', label='preview-spec')\n",
+    "    axs[1].plot(wavelength, error, color='k')\n",
+    "    axs[2].plot(wavelength, snr, color='k')\n",
+    "    \n",
+    "    # Then use a for loop to add the level 2 products on top\n",
+    "    for prod in level2:\n",
+    "        # level 2 product data\n",
+    "        prod_wavelength, prod_flux, prod_error, prod_snr, prod_gratings = get_non_variable_data(prod) \n",
+    "\n",
+    "        axs[0].plot(prod_wavelength, prod_flux, alpha=0.8, label=prod_gratings)\n",
+    "        axs[1].plot(prod_wavelength, prod_error, alpha=0.8)\n",
+    "        axs[2].plot(prod_wavelength, prod_snr, alpha=0.8) \n",
+    "    \n",
+    "    # Add all the labels & limits\n",
+    "    plt.suptitle(f'{targ.upper()} cspec')\n",
+    "    axs[0].set_ylabel('Flux\\n[ergs/s/cm^2/$\\mathrm{\\AA}$]')\n",
+    "    axs[1].set_ylabel('Error\\n[ergs/s/cm^2/$\\mathrm{\\AA}$]')\n",
+    "    axs[2].set_ylabel('SNR')\n",
+    "    plt.xlabel('Wavelength [$\\mathrm{\\AA}$]')\n",
+    "    axs[0].legend(ncol=len(level2)+1, loc='upper left')\n",
+    "\n",
+    "    axs[0].set_xlim(wavelength.min(), wavelength.max())\n",
+    "    if ull_targ == 'v-xx-cha':\n",
+    "        axs[0].set_ylim(-1E-15, 5E-14)\n",
+    "        axs[1].set_ylim(-1E-15, 5E-15)\n",
+    "    else:\n",
+    "        # set the limit based on the preview-spec to avoid grating edges\n",
+    "        axs[0].set_ylim(np.nanmin(flux), np.nanmax(flux)) \n",
+    "        axs[1].set_ylim(np.nanmin(error), np.nanmax(error)) \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
```

Comment:
It might help to add some commentary on the plots here.

Reply:
@jotaylor I've added some text below the cspec and aspec plots. Let me know if they sound ok!

Reply:
These sound good to me

Reply:
I fixed a typo in one and replaced this sentence 
"so the more indepth notebook shows how to use the ULLYSES coadd and abut code to create your own products"
with "For this reason, the ULLYSES team recommends using the individual `cspec` files for scientific purposes.". We don't actually support custom abutment right now 😬